### PR TITLE
Feat/shib auth

### PR DESCRIPTION
**Pull request by anton-kachurin**
Adds `shib-auth` command that allows to authenticate your CLI with Shibboleth and Duo 2FA

Based on this approach:
https://aws.amazon.com/blogs/security/how-to-implement-federated-api-and-cli-access-using-saml-2-0-and-ad-fs/

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -14,16 +16,21 @@ RUN set -ex; \
...
-# Install aws-shell (which also install aws-cli)
-RUN pip install --upgrade \
+# Install aws-shell (which also installs aws-cli) and some dependencies
+RUN pip3 install --upgrade \
       pip \
       aws-shell \
-      awsebcli;
+      awsebcli \
+      boto==2.49.0 \
+      pyppeteer==0.0.25
```
Comment (bfenster) [inline Dockerfile:34]: Is there a semicolon required here?

File: auth.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+        with open(filename, 'w+') as configfile:
+            config.write(configfile)
+
+    browser = await launch(
+        headless=True,
+        executablePath="/usr/bin/chromium-browser",
+        args=['--no-sandbox', '--disable-gpu']
+    )
+    page = await browser.newPage()
+    await page.goto('http://bu.edu/awslogin')
```
Comment (bfenster) [inline auth.py]: `https://www.bu.edu/awslogin` ?

File: Dockerfile (hunk 1 above)
Comment (anton-kachurin) [reply Dockerfile:34]: It's not required, the image builds just fine

File: auth.py (hunk 2 above)
Comment (anton-kachurin) [reply auth.py]: Actually, I want to make it a standalone parameter. And yes, I'll make it the one you suggest. Thanks!

**Pull request merged**